File: bindings/csharp/src/ProvenDds.cs
// SPDX-License-Identifier: PMPL-1.0-or-later
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// DDS protocol types for proven-servers.

namespace Proven;

/// <summary>ReliabilityKind matching the Idris2 ABI tags (0-1).</summary>
public enum ReliabilityKind : byte
{
    BestEffort = 0,
    Reliable = 1
}

/// <summary>DurabilityKind matching the Idris2 ABI tags (0-2).</summary>
public enum DurabilityKind : byte
{
    TransientLocal = 0,
    Transient = 1,
    Persistent = 2
}

/// <summary>HistoryKind matching the Idris2 ABI tags (0-1).</summary>
public enum HistoryKind : byte
{
    KeepLast = 0,
    KeepAll = 1
}

/// <summary>OwnershipKind matching the Idris2 ABI tags (0-1).</summary>
public enum OwnershipKind : byte
{
    Shared = 0,
    Exclusive = 1
}

/// <summary>EntityType matching the Idris2 ABI tags (0-5).</summary>
public enum EntityType : byte
{
    Participant = 0,
    Publisher = 1,
    Subscriber = 2,
    Topic = 3,
    DataWriter = 4,
    DataReader = 5
}

/// <summary>ParticipantState matching the Idris2 ABI tags (0-4).</summary>
public enum ParticipantState : byte
{
    Idle = 0,
    Joined = 1,
    Publishing = 2,
    Subscribing = 3,
    Leaving = 4
}

File: bindings/php/src/ProvenDds.php
<?php
// SPDX-License-Identifier: PMPL-1.0-or-later
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// DDS protocol types for proven-servers.

declare(strict_types=1);

namespace ProvenServers;

/** ReliabilityKind matching the Idris2 ABI tags. */
enum ReliabilityKind: int
{
    case BestEffort = 0;
    case Reliable = 1;
}

/** DurabilityKind matching the Idris2 ABI tags. */
enum DurabilityKind: int
{
    case TransientLocal = 1;
    case Transient = 2;
    case Persistent = 3;
}

/** HistoryKind matching the Idris2 ABI tags. */
enum HistoryKind: int
{
    case KeepLast = 0;
    case KeepAll = 1;
}

/** OwnershipKind matching the Idris2 ABI tags. */
enum OwnershipKind: int
{
    case Shared = 0;
    case Exclusive = 1;
}

/** EntityType matching the Idris2 ABI tags. */
enum EntityType: int
{
    case Participant = 0;
    case Publisher = 1;
    case Subscriber = 2;
    case Topic = 3;
    case DataWriter = 4;
    case DataReader = 5;
}

/** ParticipantState matching the Idris2 ABI tags. */
enum ParticipantState: int
{
    case Idle = 0;
    case Joined = 1;
    case Publishing = 2;
    case Subscribing = 3;
    case Leaving = 4;
}
